src/vmstats_server.erl

```diff
@@ -169,9 +132,38 @@ sched_time_available() ->
         error:badarg -> false
     end.
 
--spec base_key() -> term().
-base_key() ->
-    case application:get_env(vmstats, base_key) of
-        {ok, V} -> V;
-        undefined -> "vmstats"
-    end.
+write_io_stats(Sink, Key, #state{prev_io = {PrevIn, PrevOut}}) ->
+    {{input, In}, {output, Out}} = erlang:statistics(io),
+    Sink:increment([Key, "bytes_in"], In - PrevIn),
+    Sink:increment([Key, "bytes_out"], Out - PrevOut),
+    {In, Out}.
+
+write_gc_stats(Sink, Key, #state{prev_gc = {PrevGCs, PrevWords}}) ->
+    {GCs, Words, _} = erlang:statistics(garbage_collection),
+    Sink:increment([Key, "count"], GCs - PrevGCs),
+    Sink:increment([Key, "words_reclaimed"], Words - PrevWords),
+    {GCs, Words}.
+
+%% There are more options available, but not all were kept.
+%% Memory usage is in bytes.
+write_memory_stats(Sink, Key) ->
+    Sink:gauge([Key, "total"], erlang:memory(total)),
+    Sink:gauge([Key, "procs_used"], erlang:memory(processes_used)),
+    Sink:gauge([Key, "atom_used"], erlang:memory(atom_used)),
+    Sink:gauge([Key, "binary"], erlang:memory(binary)),
+    Sink:gauge([Key, "ets"], erlang:memory(ets)).
```

Comment:
@ferd would you prefer for a possible `vmstats_statsderl` sink to be directly included in `vmstats`, in `statsderl` or as a separate project? I agree with @lexmag's preference, which is to make a separate `vmstats_statsderl` project (so that we don't clutter neither `vmstats` nor `statsder`), but I want your opinion as well of course :).